tutorials/Multi_objective_multi_fidelity_BO.ipynb

```diff
@@ -0,0 +1,1720 @@
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Model Initialization ###\n",
+    "We use a multi-output SingleTaskGP to model the problem with a homoskedastic Gaussian likelihood with an inferred noise level. \n",
+    "The model is initialized with 5 random points where the fidelity dimension of the initial points is sampled from a distribution : $p(s)=C(s)^{-1}$ "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 6,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from botorch.models.gp_regression import SingleTaskGP\n",
+    "from botorch.models.transforms.outcome import Standardize\n",
+    "from gpytorch.mlls.exact_marginal_log_likelihood import ExactMarginalLogLikelihood\n",
+    "\n",
+    "exp_arg = torch.tensor(4)\n",
+    "\n",
+    "\n",
+    "def gen_init_data(dim_x, points, dim_y):\n",
+    "    # Generates training data with Fidelity dimension sampled from a probability distribution that depends on Cost function\n",
+    "    train_x = torch.rand(points, dim_x)\n",
+    "    fid_samples = torch.linspace(0, 1, 101)\n",
+    "    prob = 1 / cost_func(\n",
+    "        fid_samples, exp_arg\n",
+    "    )  # Probability calculated from the Cost function\n",
+    "    prob = prob / torch.sum(prob)  # Normalizing\n",
+    "    idx = prob.multinomial(\n",
+    "        num_samples=points, replacement=True\n",
+    "    )  # Generating indices to choose fidelity samples\n",
+    "    train_x[:, -1] = fid_samples[idx]\n",
+    "    train_x = train_x.to(**tkwargs)\n",
+    "    train_obj = problem(\n",
+    "        train_x, dim_y\n",
+    "    )  # Calls the Problem wrapper to generate train_obj\n",
+    "    return train_x, train_obj\n",
+    "\n",
+    "\n",
+    "def initialize_model(train_x, train_obj):\n",
+    "    # Initializes a SingleTaskGP with Matern 5/2 Kernel and returns the model it's MLL\n",
+    "    model = SingleTaskGP(\n",
+    "        train_x, train_obj, outcome_transform=Standardize(m=train_obj.shape[-1])\n",
+    "    )\n",
+    "    mll = ExactMarginalLogLikelihood(model.likelihood, model)\n",
+    "    return mll, model"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Helper functions to define Cost ###\n",
+    "\n",
+    "The cost_func function returns an exponential cost from the fidelity. The Cost_callable is a wrapper around it that takes care of the input output shapes. This is given as a callable function to MOMF that internally divides the hypervolume by cost."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 7,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Min Cost: tensor(1.)\n",
+      "Max Cost: tensor(54.5981)\n"
+     ]
+    }
+   ],
+   "source": [
+    "def cost_func(x, A):\n",
+    "    # A simple exponential cost function with the functionality of working on Numpy arrays instead of tensors if required\n",
```

Comment:
How would this work on np arrays?

Reply:
I removed the code for the np arrays but missed updating the comment. Corrected now